### DISCUSSION TITLE
.split(".") does not work on the same case where .split("chr") or .split("_") are working

### DISCUSSION THREAD

**Discussion by kubranarci** [nextflow-io/nextflow#4258] [Q&A] [answered]
I am trying to map a list using chromosome names inside such: 

[/filepath/ALL.chr1.GRCh38.phased.CHR.vcf]
[/filepath/ALL.chr2.GRCh38.phased.CHR.vcf]
[/filepath/ALL.chr3.GRCh38.phased.CHR.vcf]

to 

[/filepath/ALL.chr1.GRCh38.phased.CHR.vcf, chr1]
[/filepath/ALL.chr2.GRCh38.phased.CHR.vcf, chr2]
[/filepath/ALL.chr3.GRCh38.phased.CHR.vcf, chr3]

This process would be easy if I could just split by "." but it does not work. Anyone has any idea?

**Comment by kubranarci** [comment, marked as answer]
Answer from Marcel: 

```groovy
Channel
  .fromPath("data/*")
  .set { my_ch }

my_ch.view()

my_ch
  .map { path -> [path, path.name.split("\\.")[1]] }
  .view()
```

**Reply by mribeirodantas** [reply]
🥳

**Comment by kubranarci** [comment]
my solution: 

```groovy
intervals_ch = [chr1, chr2, chr3]
files
        .combine(intervals_ch)
        .filter{it[0].baseName.contains(it[1]+".")}
        .set{new_ch}
```
